### DISCUSSION TITLE
Adding request in response

### DISCUSSION THREAD

**Discussion by lucascalsilva** [mockoon/mockoon#537] [Help] [answered]
Hi,

I have a mocking API that receives some data by post:
`{
    "requestor":"1234",
    "customer":"customer"
}`

In the response, I wanted to use the existing data in the request, but I wanted to add an additional property:

`{
    "incidentId": "b6d1b1b9-08e5-4e35-8121-325042daa7ec",
    "requestor":"1234",
    "customer":"customer"
}`

The thing is, I couldn't find a way to add a property to the existing request body in the response.

I know that I could do this manually, as follows:
`{
  "incidentId": "{{faker 'random.uuid'}}",
  "requestor": {{body 'requestor'}},
  "customer" :{{body 'customer' 'customer' true}}
}`

The issue is, since this payload can be quite big, I wanted to figure out a way to do this dynamically. Is it even possible?

**Comment by 255kb** [comment, marked as answer]
Hi @lucascalsilva 
There is no way to "mutate" an entering object yet. You will have to recreate the JSON manually and use the [`body` helper](https://mockoon.com/docs/latest/templating/mockoon-request-helpers/#body) to reuse the request's body properties.
